backend/model/Users.js

```diff
@@ -6,103 +6,222 @@ require("firebase/auth");
 
 // Firestore
 const db = firebase.firestore();
-const users = db.collection("users");
+const usersCollection = db.collection("users");
 
 const UsersAPI = {
   // FOR TESTING PURPOSES ONLY, NOT A FUNCTION
   addUser: function (username, first, last) {
-    users.doc(username).set({
+    usersCollection.doc(username).set({
       first: first,
       last: last,
     });
   },
-  // user: firebase.auth().currentUser,
-
-  createNewUser: function (username, firstName, lastName) {
-    // same functionality
-    this.updateUserProfile(username, firstName, lastName);
-    console.log(user);
-    /*
-    this.user.updateProfile({displayName: username});
-    users.doc(user.uid).set({
-      first_name: firstName,
-      last_name: lastName
-    });
-    */
-  },
 
-  uploadUserProfilePicture: function () {
-    // TODO this function may not work correctly.
-    // create a cloud storage refrence
-    var storageRef = firebase
-      .storage()
-      .ref(user.uid + "/profilePicture/" + file.name);
-
-    // save to cloud storage
-    var task = storageRef.put(file);
+  // Profile pic, friends, hugs, chatrooms, corkboard_id
+  /*
+   * @Param: firebase::auth::User current_user - Current signed in user from firebase authentication
+   */
+  createNewUser: async function (current_user) {
+    let created = false;
+
+    // initialize local object containing initial user values
+    const user = {
+      user_id: current_user.uid,
+      username: "",
+      first_name: "",
+      last_name: "",
+      profile_pic: "",
+      day_hug_count: 0,
+      current_streak: 0
+    };
 
-    // update user's photo URL to the saved cloud storage url
-    user.updateProfile({ photoURL: storageRef.getDownloadURL() });
+    await usersCollection
+      .doc(current_user.uid)
+      .set(user) // set uid document to new user values
+      .then(() => {
+        console.log(`User created with ID: ${current_user.uid}`);
+        created = true;
+      })
+      .catch((error) => {
+        console.log(`Error adding user: ${error}`);
+        created = false;
+      });
+
+    return created;
   },
 
-  getUserProfile: function (userId) {
-    var userProfile = {
-      username: this.user.username,
-      firstName: users.get(user.first_name),
-      lastName: users.get(user.last_name),
-      profPicUrl: this.user.photoURL,
-    };
-    return userProfile;
+  // returns users profile information in an object
+  // takes in signed-in user 
+  getUserProfile: async function (current_user) {
+
+    var userDocRef = usersCollection.doc(current_user.uid);
+    var userProfile;
+
+    // access document
+    await userDocRef.get().then(function(userDoc) {
+      if (userDoc.exists) {
+        // set userProfile to retrieved data
+        // not sure this is how to retrieve data
+        userProfile = {
+          username:     userDoc.get("username"),
```

Comment:
Just curious but userDoc.get("username"), is that the same thing as userDoc.data().username? That's what Tyus and I figured out on Saturday.

Reply:
Both should be fine

Reply:
https://firebase.google.com/docs/reference/js/firebase.firestore.DocumentSnapshot#data